app/controllers/categories_controller.rb

```diff
@@ -2,14 +2,25 @@
 
 class CategoriesController < ApplicationController
   include CollectionPageSetupConcern
+  include CategoriesHelper
   include Shared
   before_action :set_category, only: %i[show edit update destroy]
+  respond_to :json, only: [:index]
 
   # GET /categories
   # GET /categories.json
 
   def index
     page_collection('@categories', 'category')
+    respond_to do |format|
```

Comment:
We might want to create a helper for this as a lot of similar code appears in other files.